common/devdoc/gballoc_ll_passthrough.md

```diff
@@ -0,0 +1,78 @@
+# gballoc_ll_passthrough requirements
+================
+
+## Overview
+
+gballoc_ll_passthrough is a module that delegates all call of its APIs to the ones from C standard lib.
+
+## References
+
+
+## Exposed API
+
+```c
+    MOCKABLE_FUNCTION(, int, gballoc_ll_init, void*, params);

```

Comment:
>gballoc_ll_init [](start = 29, length = 15)

This design assumes forces implementors and users to have one heap for the entire process. IMO, the interface should support creation and usage of multiple heaps. I'm thinking malloc, free, calloc, realloc can take as input a heap handle as well. If it is null, they can use the default heap.

Reply:
"we" will have 1 heap per process. With this interface, that's all we're going to get. We might get more "heaps" in a later iteration of the code. Added a task to consider it

"others" are free to create/to not create their own heaps. Task 7745659: Consider adding "heap" as parameter to all allocations in gballoc_ll/hl

---
In reply to: [457685081](https://github.com/Azure/azure-c-pal/pull/11#discussion_r457685081) [](ancestors = 457685081)